app/Http/Controllers/IncidentController.php

```diff
@@ -33,25 +33,54 @@ public function active(Request $request): JsonResponse
     public function search(IncidentSearchRequest $request): JsonResponse
     {
         $day = $request->get('day');
+        $before = $request->get('before');
+        $after = $request->get('after');
+
+        if($day && ($before || $after)){
+            abort(422);
+        }
+
+        $timeRange = false;
+        if($after){
+            $after = new Carbon($after);
+            $before = $before ? new Carbon($before) : new Carbon();
```

Comment:
We can simplify, here, a little:
```suggestion
            $before = new Carbon($before)
```

`$before` will be initialized either with a value or with null, due to how `$request->get()` works.